src/main/java/org/itsallcode/openfasttrace/importer/legacytag/LegacyTagImporterFactory.java

```diff
@@ -39,43 +40,61 @@
 
     private static LegacyTagImporterConfig defaultConfig = new LegacyTagImporterConfig();
 
-    private final List<PathConfig> pathConfigs;
+    private final Supplier<LegacyTagImporterConfig> config;
 
     public LegacyTagImporterFactory()
     {
-        this(defaultConfig.getPathConfigs());
+        this(() -> defaultConfig);
     }
 
-    public LegacyTagImporterFactory(final List<PathConfig> pathConfigs)
+    public LegacyTagImporterFactory(final LegacyTagImporterConfig config)
     {
-        this.pathConfigs = pathConfigs;
+        this(() -> config);
+    }
+
+    public LegacyTagImporterFactory(final Supplier<LegacyTagImporterConfig> config)
+    {
+        this.config = config;
     }
 
     @Override
-    public boolean supportsFile(final Path file)
+    public boolean supportsFile(final Path path)
     {
-        return findConfig(file).isPresent();
+        return findConfig(path).isPresent();
     }
 
     private Optional<PathConfig> findConfig(final Path file)
     {
-        return this.pathConfigs.stream().filter(config -> config.matches(file)).findFirst();
+        final Path relativePath = makeRelative(file);
+        return this.config.get().getPathConfigs().stream() //
+                .filter(config -> config.matches(relativePath)) //
+                .findFirst();
```

Comment:
What happens if multiple match?

Reply:
Wouldn't a hash map be more efficient here than iterating all path configs with each find?

Reply:
* For multiple matches we take the first matching path.
* We can't use a map because the matching logic is based on Java's path matcher.